### PR TITLE
Adjust AllGather, AlltoAll count

### PR DESCRIPTION
**Pull request by lappazos**
## What
Changes AllGather dst.count for the number of elements in the entire dst buffer, for all ranks combined.
Changes AlltoAll dst.count & src.count for the number of elements in the entire src & dst buffers, for all ranks combined.

## Why ?
API Change in PR #225 

## How ?
Change implementation in TL and use cases in tests.

### CODE REVIEW

File: src/components/tl/nccl/tl_nccl_coll.c [hunk 1]
```diff
@@ -103,7 +103,7 @@ ucc_status_t ucc_tl_nccl_alltoall_start(ucc_coll_task_t *coll_task)
     ucc_rank_t peer;
 
     task->super.super.status = UCC_INPROGRESS;
-    data_size = (size_t)task->args.src.info.count *
+    data_size                = (size_t)(task->args.src.info.count / gsize) *
```
Comment (vspetrov) [inline src/components/tl/nccl/tl_nccl_coll.c:106]: plz add ucc_assert(task->args.src.info.count % gsize == 0) at line 108

**Review by vspetrov — APPROVED** [review]

**Review by Sergei-Lebedev — APPROVED** [review]

**Review by manjugv — APPROVED** [review]

**Comment by manjugv** [comment]
Ready to merge?

**Comment by lappazos** [comment]
Failed in torch alltoall test - expected due to the API change. update torch is in https://github.com/facebookresearch/torch_ucc/pull/21

**Comment by lappazos** [comment]
@vspetrov @Sergei-Lebedev @manjugv please merge

**Pull request merged**